### PR TITLE
feat: publish stac to odr bucket TDE-1042

### PR DESCRIPTION
**Pull request by paulfouquet**
#### Motivation

In order to continuously publish data in the ODR bucket nz-elevation, the publish workflow needs to be updated to point to nz-elevation rather than the current bucket linz-elevation.

#### Modification

The publish workflow has been duplicated from https://github.com/linz/imagery/blob/7c55691c40e0b6b7d0d578c9fed2e9f7f7ce9b36/.github/workflows/publish.yml
GH environment prod has been created along with secrets.

This change should not be deployed before the elevation data has been re-processed into the linz-elevation bucket.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -8,6 +8,46 @@ jobs:
...
+      - name: Validate STAC Catalog
+        uses: docker://ghcr.io/linz/argo-tasks:v3
+        with:
+          args: stac-validate /github/workspace/stac/catalog.json
+
+      - name: Validate STAC Collections
+        run: |
+          # Enable double star operator
+          shopt -s globstar
+          docker run -v "$PWD:$PWD" ghcr.io/linz/argo-tasks:v3 stac-validate "$PWD"/stac/**/collection.json
```
Comment (l0b0) [inline .github/workflows/publish.yml:33]: Would it make sense for this to use the same pattern as line 39?
```suggestion
          docker run --volume="$PWD:/repo" ghcr.io/linz/argo-tasks:v3 stac-validate /repo/stac/**/collection.json
```

Ditto below.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -30,19 +70,25 @@ jobs:
...
           aws-region: ap-southeast-2
           mask-aws-account-id: true
-          role-to-assume: ${{ secrets.AWS_CI_ROLE }}
+          role-to-assume: ${{ secrets.AWS_ODR_DATA_MANAGER_ROLE }}
+          role-chaining: true
 
       # Sync STAC files only on push to 'master'
       - name: Sync STAC
         uses: docker://ghcr.io/linz/argo-tasks:v2
-        if: github.ref == 'refs/heads/master'
```
Comment (l0b0) [inline .github/workflows/publish.yml:46]: This probably means we should remove the comment above the step.

**Review by l0b0 — COMMENTED** [review]
There's still a few "linz-elevation" mentions outside the `stac` directory; should any more of those be changed?

**Review by l0b0 — DISMISSED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (l0b0) [reply .github/workflows/publish.yml:33]: We can do this later.

**Review by l0b0 — APPROVED** [review]

**Review by MDavidson17 — APPROVED** [review]

**Review by blacha — APPROVED** [review]

**Pull request merged**